### PR TITLE
Implement noexpandtab for `>>`, solve issue #795

### PR DESCRIPTION
**Pull request by ajylee**
- Add option `expandtab` with default YES.
- Use tab instead of spaces in `xvim_shift:right:withMotionPoint:count`
  for `right:YES` if `expandtab` is NO.
- Solves issue #795

### CODE REVIEW

**Comment by steinnes** [comment]
Works for me! (sorry how long it took me to test!).  Thanks!

**Comment by steinnes** [comment]
I hope this gets merged, I closed my issue at least.

File: XVim/Test/XVimTestCase.m [hunk 1]
```diff
@@ -68,6 +69,7 @@ - (void)setUp{
     [[[XVimLastActiveSourceView() xvim_window] sourceView] xvim_changeSelectionMode:XVIM_VISUAL_NONE];
     [XVimLastActiveSourceView() setString:self.initialText];
     [XVimLastActiveSourceView() setSelectedRange:self.initialRange];
+    [XVim instance].options.expandtab = YES;
```
Comment (JugglerShu) [inline XVim/Test/XVimTestCase.m]: Do you need this here? This setup method is commonly used by all the test cases. I don't want to put test case specific setup here.

Comment (ajylee) [inline XVim/Test/XVimTestCase.m]: I will see if I can remove it.

I was not sure if options get reset to default between tests. If they are reset to default, then it is not necessary.

**Comment by ajylee** [comment]
I removed the change to `XVim/Test/XVimTestCase.m` and instead used `:set expandtab` from inside the test cases.

**Pull request merged**